src/common/scaled-scene-buffer.c

```diff
@@ -177,6 +220,10 @@ scaled_scene_buffer_create(struct wlr_scene_tree *parent,
 	self->active_scale = 1;
 	self->drop_buffer = drop_buffer;
 	wl_list_init(&self->cache);
+	if (!impl->cached_buffers.next) {
```

Comment:
It is not clear to me what this does.

What does `!impl->cached_buffers.next` mean? Couldn't it mean that the list has never been initialized OR contains items. Feels like red flag to me.

Reply:
It basically just means that its calloc'd and was not initialized. An empty list points to itself for both `.prev` and `.next` and a list with items points to these items. Maybe we could add some `wl_list_maybe_init()` helper or something? Other ideas welcome, the issue here is that this is a shared list between all scaled-buffers of a given implementation (just the the scaled-font-buffer at the moment). See also my reply to the comment regarding the placement of the list itself above.